### PR TITLE
FEAT(client): Introduced new mumble API to link and unlink

### PR DESCRIPTION
**Pull request by mryamac**
This patch introduces a new API to link and unlink the given channels to the given link ids.

### Checks

- [x] My commits follow the [commit guidelines](https://github.com/mumble-voip/mumble/blob/master/COMMIT_GUIDELINES.md)

### CODE REVIEW

File: src/mumble/API_v_1_x_x.cpp [hunk 1]
```diff
@@ -1971,7 +2050,9 @@ MumbleAPI_v_1_0_x getMumbleAPI_v_1_0_x() {
 			 setMumbleSetting_string_v_1_0_x,
 			 sendData_v_1_0_x,
 			 log_v_1_0_x,
-			 playSample_v_1_0_x };
+			 playSample_v_1_0_x,
+			 requestChannelLink_v_1_3_x,
+			 requestChannelUnlink_v_1_3_x };
```
Comment (Krzmbrzl) [inline src/mumble/API_v_1_x_x.cpp:2055]: These can't appear here as these functions are not part of the interface v1.0

File: src/mumble/API_v_1_x_x.cpp [hunk 2]
```diff
@@ -2012,7 +2093,52 @@ MumbleAPI_v_1_2_x getMumbleAPI_v_1_2_x() {
 			 setMumbleSetting_string_v_1_0_x,
 			 sendData_v_1_0_x,
 			 log_v_1_0_x,
-			 playSample_v_1_2_x };
+			 playSample_v_1_2_x,
+			 requestChannelLink_v_1_3_x,
+			 requestChannelUnlink_v_1_3_x };
```
Comment (Krzmbrzl) [inline src/mumble/API_v_1_x_x.cpp:2098]: Nor are they part of v1.2

File: plugins/MumblePlugin.h [hunk 3]
```diff
@@ -1807,6 +1808,32 @@ struct MUMBLE_API_STRUCT_NAME {
...
+	 * Requests Mumble to unlink the given channel to the given unlinkID.
+	 *
+	 * @param callerID The ID of the plugin calling this function
+	 * @param channelID The ID of the channel to unlink
+	 * @param unlinkID The ID of the base channel to unlink to
+	 * @returns The error code. If everything went well, STATUS_OK will be returned.
+	 */
+	mumble_error_t(MUMBLE_PLUGIN_CALLING_CONVENTION *requestChannelUnlink)(mumble_plugin_id_t callerID,
+																		   mumble_channelid_t channelID,
+																		   mumble_channelid_t unlinkID);
```
Comment (Krzmbrzl) [inline plugins/MumblePlugin.h:1836]: These should go at the end of the group of the other `request*` functions (in order to make it easier to find functions).
Plus, we'll want to only define these functions if the API version is >= 1.3 - aka: wrap this in an `#if` similar to the one used for the definition of the `PARAM_v1_2` macro.

File: plugins/MumblePlugin.h [hunk 4]
```diff
@@ -129,7 +129,7 @@
  * MUMBLE_QUOTE
  * Quotes the given value WITHOUT macro expansion
  */
-#	define MUMBLE_QUOTE(arg) #    arg
+#	define MUMBLE_QUOTE(arg) # arg
```
Comment (Krzmbrzl) [inline plugins/MumblePlugin.h:132]: The formatting CI job will probably complain about this change (likely introduced by using a different clang-format version that v10)

File: src/mumble/API_v_1_x_x.cpp (hunk 1 above)
Comment (mryamac) [reply src/mumble/API_v_1_x_x.cpp:2055]: I tried to remove these definitions, but the compiler gave an error. Also, you gave me the answer, and I can solve it using `#if` .

File: src/mumble/API_v_1_x_x.cpp (hunk 2 above)
Comment (mryamac) [reply src/mumble/API_v_1_x_x.cpp:2098]: got it.

File: plugins/MumblePlugin.h (hunk 4 above)
Comment (mryamac) [reply plugins/MumblePlugin.h:132]: I changed it to the previous version.

File: src/mumble/API.h [hunk 5]
```diff
@@ -91,14 +91,14 @@ public slots:
 								mumble_channelid_t channelID, const char **name,
 								std::shared_ptr< api_promise_t > promise);
 	void getAllUsers_v_1_0_x(mumble_plugin_id_t callerID, mumble_connection_t connection, mumble_userid_t **users,
-							 size_t *userCount, std::shared_ptr< api_promise_t > promise);
+							 std::size_t *userCount, std::shared_ptr< api_promise_t > promise);
```
Comment (Krzmbrzl) [inline src/mumble/API.h:94]: This change is unrelated to your PR, no? If so, please remove it (and others like it).
Or put them in a separate commit - that'd also be fine.

**Review by Krzmbrzl — COMMENTED** [review]
The placement of the new API functions in the `MumblePlugin.h` header is still off but otherwise this looks good.

File: src/mumble/API.h (hunk 5 above)
Comment (Hartmnt) [reply src/mumble/API.h:94]: This is #6329

@mryamac If you rebase the branch on the current master (after the merge of #6329) it should be fine

Comment (mryamac) [reply src/mumble/API.h:94]: While squashing commits, I made a mistake, including the last master commit. That's why some std headers still exist. I will create a new PR and replace the new API functions in that PR.

Comment (Hartmnt) [reply src/mumble/API.h:94]: As I said, a ``git rebase master`` should fix this (assuming your local master on your computer is up to date)
If you want to try to improve your git skills, you could try that :)
You can also always look up git help on the internet if you have trouble. Most operations/situations are fixable with git
